packages/material-ui-styles/src/styled/styled.d.ts

```diff
@@ -14,7 +14,7 @@ export type ComponentCreator<C extends React.ElementType> = <Theme, Props extend
   options?: WithStylesOptions<Theme>,
 ) => React.ComponentType<
   Omit<JSX.LibraryManagedAttributes<C, React.ComponentProps<C>>, 'classes' | 'className'> &
-    StyledComponentProps<'root'> & { className?: string }
+    StyledComponentProps<'root'> & { className?: string } & Partial<Props>
```

Comment:
I think this should intersect with the required props. If the style function requires some props it should mark the prop as required on the component.

Reply:
Just getting rid of the `Partial` should do the trick?

Reply:
And the failing test?